pkg/client/unversioned/request.go

```diff
@@ -587,6 +590,24 @@ func (r *Request) Watch() (watch.Interface, error) {
 	return watch.NewStreamWatcher(watchjson.NewDecoder(resp.Body, r.codec)), nil
 }
 
+// updateUrlMetrics is a convenience function for pushing metrics.
+// It also handles corner cases for incomplete/invalid request data.
+func updateUrlMetrics(req *Request, resp *http.Response, err error) {
+	url := "none"
+	// In unit tests, weird urls can occur.
+	if req.baseURL != nil {
+		url = req.baseURL.Host
```

Comment:
this gives us granularity of information which is sufficient to solve the usecase which is most intersting for now, which is  backoff  for a host+port.... no doubt that there are other usecases where more lables might be useful.